src/test/java/gov/healthit/chpl/aqa/stepDefinitions/SurveillanceComplaintsSteps.java

```diff
@@ -0,0 +1,133 @@
+            final String acbComplaintId, final String oncComplaintId, final String complainantType,
+            final String summary) {
+        SurveillanceComplaintsPage.oncACBField(getDriver()).sendKeys(acb);
+        SurveillanceComplaintsPage.receivedDate(getDriver()).sendKeys(receivedDate);
+        SurveillanceComplaintsPage.closedDate(getDriver()).sendKeys(closedDate);
+        SurveillanceComplaintsPage.oncACBComplaintId(getDriver()).sendKeys(acbComplaintId);
+        SurveillanceComplaintsPage.oncComplaintId(getDriver()).sendKeys(oncComplaintId);
+        SurveillanceComplaintsPage.complainantType(getDriver()).sendKeys(complainantType);
+        SurveillanceComplaintsPage.summary(getDriver()).sendKeys(summary);
+        ((JavascriptExecutor) getDriver()).executeScript("arguments[0].click();", SurveillanceComplaintsPage.complainantContactedCheckbox(getDriver()));
+        ((JavascriptExecutor) getDriver()).executeScript("arguments[0].click();", SurveillanceComplaintsPage.developerContactedCheckbox(getDriver()));
+        ((JavascriptExecutor) getDriver()).executeScript("arguments[0].click();", SurveillanceComplaintsPage.oncATLContactedCheckbox(getDriver()));
+        ((JavascriptExecutor) getDriver()).executeScript("arguments[0].click();", SurveillanceComplaintsPage.flagForONCReviewCheckbox(getDriver()));
+    }
+
+    @When("^I click save button$")
+    public void clickSaveButton() {
+        ((JavascriptExecutor) getDriver()).executeScript("arguments[0].click();", SurveillanceComplaintsPage.saveButtonOnNewComplaintForm(getDriver()));
+        getWait().until(ExpectedConditions.visibilityOf(SurveillanceComplaintsPage.complaintsTable(getDriver())));
+    }
+
+    @When("^I click save button on Edit complaint form$")
+    public void clickSaveButtononEditComplaint() {
+        WebElement button = SurveillanceComplaintsPage.saveButtonOnEditComplaintForm(getDriver());
+        ((JavascriptExecutor) getDriver()).executeScript("arguments[0].scrollIntoView(true);", button);
+        getWait().until(ExpectedConditions.visibilityOf(SurveillanceComplaintsPage.saveButtonOnEditComplaintForm(getDriver())));
+        SurveillanceComplaintsPage.saveButtonOnEditComplaintForm(getDriver()).click();
+        getWait().until(ExpectedConditions.visibilityOf(SurveillanceComplaintsPage.complaintsTable(getDriver())));
+    }
+
+    /**
+     * Edit a complaint.
+     * @param acbComplaintId is a ACB complaint ID of a complaint to be edited
+     */
+    @When("^I click Edit complaint for ONC-ACB Complaint ID \"([^\"]*)\"$")
+    public void clickEditComplaint(final String acbComplaintId) {
+        getDriver().navigate().refresh();
+        getWait().until(ExpectedConditions.visibilityOf(SurveillanceComplaintsPage.complaintsTable(getDriver())));
+
+        List<WebElement> acbCmomplaintIdValue = getDriver().findElements(By.xpath("//*[@id=\"main-content\"]/div/div/div/div/table/tbody/tr/td[4]"));
```

Comment:
Suggest requesting an id or class or something for these; this xpath is way too deep

...later

Wait, why is a Step so concerned with xpaths? Can this be a Page Element that the step checks?

Reply:
I was going to ask for Ids, but i had worked on this part on weekend, so i used what was there. I'll request a better locator. I tried adding it as a page element and using with list and it wouldn't work at that time, but was running fine with xpaths, not sure why - selenium glitch maybe.